File: src/java/nz/co/searchwellington/commentfeeds/detectors/DateRegexCommentFeedDetector.java
package nz.co.searchwellington.commentfeeds.detectors;

import java.util.regex.Matcher;
import java.util.regex.Pattern;

public class DateRegexCommentFeedDetector implements CommentFeedDetector {

	static Pattern yearMonthDateRegex = Pattern.compile(".*\\d{4}/\\d{2}/\\d{2}.*");
	
    public boolean isValid(String url) {
    	Matcher matcher = yearMonthDateRegex.matcher(url);
    	return url != null && matcher.matches();
    }
    
}

File: src/main/scala/nz/co/searchwellington/commentfeeds/detectors/DateRegexCommentFeedDetector.scala
package nz.co.searchwellington.commentfeeds.detectors

import nz.co.searchwellington.model.Resource

import java.net.URL
import java.util.regex.Pattern

class DateRegexCommentFeedDetector extends CommentFeedDetector {

  private val yearMonthDateRegex = {
    Pattern.compile(""".*\d{4}/\d{2}/\d{2}.*""")
  }

  override def isValid(url: URL, source: Resource): Boolean = {
    yearMonthDateRegex.matcher(url.toExternalForm).matches
  }

}